src/main.c

```diff
@@ -351,13 +356,75 @@ key_binding_func_key (MBWindowManager   *wm,
   hd_dbus_send_event (s);
 }
 
+/* Fill in the array with input devices type is touchscreen */
+static void
+enumerate_input_devices (Display *dpy)
+{
+  static Atom atom_touchscreen = None;
+
+  if (atom_touchscreen == None)
+    atom_touchscreen = XInternAtom (dpy, XI_TOUCHSCREEN, True);
+
+  if (atom_touchscreen != None)
+    {
+      XDeviceInfo *devinfo;
+      int i, ndev, nclass = 0;
+      XEventClass eclass[64];
+
+      /* get XInput DeviceMotion events */
+      devinfo = XListInputDevices (dpy, &ndev);
+
+      if (ndev > sizeof(eclass))
```

Comment:
How does sizeof(eclass) corellate with the number of available devices?

Reply:
right, this should be if (ndev > sizeof(eclass) / sizeof(eclass[0])) or
if (ndev > G_N_ELEMENTS(eclass))

BTW the same should be done for if (ndev > sizeof(xdevice_is_ts))

Thanks for pointing that out